src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -1248,11 +1277,11 @@ ErrorIfQueryNotSupported(Query *queryTree)
 	const char *filterHint = "Consider using an equality filter on the distributed "
 							 "table's partition column.";
 
-	if (queryTree->hasSubLinks)
+	if (queryTree->hasSubLinks && SublinkList(queryTree) == NIL)
```

Comment:
Could you add a comment noting that why we can have hasSubLinks set but SublinkList(queryTree) empty?